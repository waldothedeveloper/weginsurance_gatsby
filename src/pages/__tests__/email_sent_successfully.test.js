import { render, screen } from "@testing-library/react";

import EmailSentSuccessfully from "../auth/email_sent_successfully";
import React from "react";
import { navigate } from "gatsby";
import userEvent from "@testing-library/user-event";

const user = userEvent.setup();
//
describe(`Email sent successfully`, () => {
  it(`Should render fallback UI if email is not provided`, () => {
    const container = render(
      <EmailSentSuccessfully location={{ state: { email: undefined } }} />
    );

    expect(container).toMatchSnapshot();
  });

  it(`should navigate back home  when Gatsby Link (anchor) is clicked`, () => {
    render(
      <EmailSentSuccessfully
        location={{ state: { email: `email not provided` } }}
      />
    );
    const link = screen.getByTestId(`back-to-home`);

    expect(link).toHaveAttribute(`href`, `/`);
  });

  it(`should render 200 number when valid email is provided`, () => {
    const { getByText } = render(
      <EmailSentSuccessfully
        location={{ state: { email: `test@gmail.com` } }}
      />
    );

    expect(getByText(`200`)).toBeInTheDocument();
  });

  it(`should navigate to gmail provider if Gatsby Link (anchor) is clicked`, async () => {
    render(
      <EmailSentSuccessfully
        location={{ state: { email: `test@gmail.com` } }}
      />
    );
    const link = screen.getByTestId(`gmail-link`);
    await user.click(link);
    await navigate(`https://mail.google.com/mail/u/0/#inbox`);

    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith(
      `https://mail.google.com/mail/u/0/#inbox`
    );
  });
});
