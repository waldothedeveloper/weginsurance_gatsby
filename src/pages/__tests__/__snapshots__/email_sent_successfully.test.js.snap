// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email sent successfully Should render fallback UI if email is not provided 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen"
      >
        <div
          class="min-h-full bg-white px-4 py-16 sm:px-6 sm:py-24 md:grid md:place-items-center lg:px-8"
        >
          <div
            class="mx-auto max-w-max"
          >
            <main
              class="sm:flex"
            >
              <p
                class="text-4xl font-extrabold text-sky-600 sm:text-5xl"
              >
                200
              </p>
              <div
                class="sm:ml-6"
              >
                <div
                  class="sm:border-l sm:border-gray-200 sm:pl-6"
                >
                  <h1
                    class="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-5xl"
                  >
                    Correo enviado con éxito!
                  </h1>
                  <p
                    class="mt-1 text-base font-semibold text-gray-500"
                  >
                    Te enviamos un correo a
                     
                    <span
                      class="font-bold text-sky-800"
                    />
                     
                    con un enlace para recuperar tu contraseña
                  </p>
                </div>
                <div
                  class="mt-10 flex space-x-3 sm:border-l sm:border-transparent sm:pl-6"
                >
                  <a
                    class="inline-flex items-center rounded-md border border-transparent bg-sky-600 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-sky-700 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                    data-testid="back-to-home"
                    href="/"
                  >
                    Regresar al inicio
                  </a>
                  <a
                    class="inline-flex items-center rounded-md border border-transparent bg-sky-100 px-4 py-2 text-sm font-medium text-sky-700 hover:bg-sky-200 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                    data-testid="gmail-link"
                    href="https://mail.google.com/mail/u/0/#inbox"
                    rel="noreferrer noopener"
                    target="_blank"
                  >
                    Llévame a mi Gmail
                  </a>
                </div>
              </div>
            </main>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen"
    >
      <div
        class="min-h-full bg-white px-4 py-16 sm:px-6 sm:py-24 md:grid md:place-items-center lg:px-8"
      >
        <div
          class="mx-auto max-w-max"
        >
          <main
            class="sm:flex"
          >
            <p
              class="text-4xl font-extrabold text-sky-600 sm:text-5xl"
            >
              200
            </p>
            <div
              class="sm:ml-6"
            >
              <div
                class="sm:border-l sm:border-gray-200 sm:pl-6"
              >
                <h1
                  class="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-5xl"
                >
                  Correo enviado con éxito!
                </h1>
                <p
                  class="mt-1 text-base font-semibold text-gray-500"
                >
                  Te enviamos un correo a
                   
                  <span
                    class="font-bold text-sky-800"
                  />
                   
                  con un enlace para recuperar tu contraseña
                </p>
              </div>
              <div
                class="mt-10 flex space-x-3 sm:border-l sm:border-transparent sm:pl-6"
              >
                <a
                  class="inline-flex items-center rounded-md border border-transparent bg-sky-600 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-sky-700 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  data-testid="back-to-home"
                  href="/"
                >
                  Regresar al inicio
                </a>
                <a
                  class="inline-flex items-center rounded-md border border-transparent bg-sky-100 px-4 py-2 text-sm font-medium text-sky-700 hover:bg-sky-200 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  data-testid="gmail-link"
                  href="https://mail.google.com/mail/u/0/#inbox"
                  rel="noreferrer noopener"
                  target="_blank"
                >
                  Llévame a mi Gmail
                </a>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
